miden-crypto/src/merkle/mmr/forest.rs

```diff
@@ -0,0 +1,245 @@
+use core::{
+    fmt::{Binary, Display},
+    ops::{BitAnd, BitOr, BitXor, BitXorAssign, ShlAssign},
+};
+
+use crate::Felt;
+
+/// A compact representation of trees (or peaks) in Merkle Mountain Range (MMR)
+///
+/// Each active bit of the stored number represents a disjoint tree with number of leaves
+/// equal to the bit position.
+///
+/// Examples:
+/// - Forest(0) is a forest with no trees.
+/// - Forest(0b01) is a forest with a single node (the smallest tree possible)
+/// - Forest(0b10) is a forest with a single binary tree with 2 leaves (3 modes)
+/// - Forest(0b11) is a forest with two trees: one with a single node, and one with 3 nodes
+/// - Forest(0b1010) is a forest with two trees: one with 8 leaves (15 nodes), one with 2 leaves (3
+///   nodes)
+#[derive(Debug, Copy, Clone, Default, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Forest(usize);
```

Comment:
In terms of naming, I wonder if we should call it `MountainRange` or `Range` (https://github.com/0xPolygonMiden/miden-node/pull/705#discussion_r1961284625)? The issue with just `Range` is that it conflicts with the `core::ops::Range` at least in name and I assume most people wouldn't think of "mountain range" when reading just "range", so it's not a very good abbreviation. Maybe `MountainRange` would be clearer, at the cost of being longer.

At least that way we'd stay consistent with the mountain metaphor and wouldn't mix tree and mountain metaphors (as pointed out in https://github.com/0xPolygonMiden/miden-node/pull/705#discussion_r1961265785).

cc @Mirko-von-Leipzig

Reply:
2c: stick with `MountainRange`; or even `MerkleMountainRange`. Typing a bit more is really not that bad with modern IDEs.

Reply:
I like `MountainRange`, makes it consistent with the rest of the project. I'd skip the `Merkle` part since:
1. Nothing in this struct touches hashing.
2. We already have the `Mmr` struct that (in my mind) combines two concepts: merkle trees and mountain ranges/forests.

I'll make the change.

Reply:
Should we also rename the `forest` fields and methods in MMR structs to `mountain_range`?

Reply:
Yes, I would make it consistent everywhere. Although there are many occurrences of "forest" in the mmr code, so to avoid having these clutter this PR, I think we should create an issue instead and do a search-replace "forest" -> "mountain_range" rename in a separate PR.
If the number of changed public APIs isn't too many, we can do just those here, e.g. `Mmr::forest` -> `Mmr::mountain_range`, and rename the internal ones later.